cmake/BLTMacros.cmake

```diff
@@ -128,10 +128,71 @@ endmacro(blt_add_target_definitions)
 
 
 ##------------------------------------------------------------------------------
+## blt_add_target_compile_flags (TO <target> FLAGS [FOO [BAR ...]])
+##
+## Adds compiler flags to a target by appending to the target's existing flags.
+##------------------------------------------------------------------------------
+macro(blt_add_target_compile_flags)
```

Comment:
Maybe.  I haven't seen a case where we would need that --  but I can see how it may be used to fix an issue with a bad actor lib or deal with some monstrosity. If you have a specific example where you know we need it, lets add it. 


That said, this is mainly used by the loop that sets up blt target deps. 
After we have imported targets for blt_reg_lib -- we will have even less control of the order of those target deps -- so one shouldn't count on prepend or append mattering.